### PR TITLE
mgmt autocent, use repo1

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

1. change package url host from central to repo1, due to incomplete index from central
2. fix version sort within same tag

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/docs/SINGLE_SERVICE_PACKAGES.md [hunk 1]
```diff
@@ -20,1564 +20,1598 @@ service
...
+    * [1.0.0-beta.9](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.9)
+    * [1.0.0-beta.8](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.8)
+    * [1.0.0-beta.7](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.7)
+    * [1.0.0-beta.6](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.6)
+    * [1.0.0-beta.5](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.5)
+    * [1.0.0-beta.4](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.4)
+    * [1.0.0-beta.3](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.3)
+    * [1.0.0-beta.2](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.2)
+    * [1.0.0-beta.1](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-datafactory/1.0.0-beta.1)
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/docs/SINGLE_SERVICE_PACKAGES.md:434]: fix version sort

File: sdk/resourcemanager/docs/SINGLE_SERVICE_PACKAGES.md [hunk 2]
```diff
@@ -20,1564 +20,1598 @@ service
...
 * [package-2021-12](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/eventgrid/resource-manager#tag-package-2021-12)
-    * [1.1.0-beta.5](https://central.sonatype.com/artifact/com.azure.resourcemanager/azure-resourcemanager-eventgrid/1.1.0-beta.5)
-    * [1.1.0-beta.4](https://central.sonatype.com/artifact/com.azure.resourcemanager/azure-resourcemanager-eventgrid/1.1.0-beta.4)
-    * [1.1.0-beta.3](https://central.sonatype.com/artifact/com.azure.resourcemanager/azure-resourcemanager-eventgrid/1.1.0-beta.3)
-    * [1.1.0](https://central.sonatype.com/artifact/com.azure.resourcemanager/azure-resourcemanager-eventgrid/1.1.0)
+    * [1.1.0](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-eventgrid/1.1.0)
+    * [1.1.0-beta.5](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-eventgrid/1.1.0-beta.5)
+    * [1.1.0-beta.4](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-eventgrid/1.1.0-beta.4)
+    * [1.1.0-beta.3](https://repo1.maven.org/maven2/com/azure/resourcemanager/azure-resourcemanager-eventgrid/1.1.0-beta.3)
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/docs/SINGLE_SERVICE_PACKAGES.md:655]: stable version before beta version

File: sdk/resourcemanager/autocent.js [hunk 3]
```diff
@@ -144,12 +144,16 @@ function writeMarkdown() {
...
+                        var aPatchNums = a.split("-beta.");
+                        var bPatchNums = b.split("-beta.");
+                        // sort GA version before beta version
                         if (aPatchNums.length < bPatchNums.length) {
                             return -1;
                         } else if (aPatchNums.length > bPatchNums.length) {
                             return 1;
+                        } else if (aPatchNums.length > 1) {
+                          // sort according to beta minor version
+                          return parseInt(bPatchNums[1]) - parseInt(aPatchNums[1]);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/autocent.js:156]: One thing I forgot... which of below is bigger?

1.2.0-beta.1
1.0.0-beta.3

Comment (XiaofeiCao) [reply sdk/resourcemanager/autocent.js:156]: Should be 1.2.0-beta.1 and appears earlier? 

It should already be dealt with from line 142 to 144, where it compares second minor version number. 

The `else` block only deal with the last minor version. 

Theoretically the following could happen, but for now we seem always to have the third version number 0:
1.0.2
1.0.1
1.0.2-beta.1

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/autocent.js:156]: ok, then they are good.

**Pull request merged**